File: frontend/src/app/api/returns.ts
import { apiRequest } from "./client";
import type { PaginatedItems } from "./products";

export type ReturnRow = {
  id: number;
  original_invoice_id: number;
  customer_id?: number | null;
  refund_method?: string;
  refund_amount?: number;
  created_at?: string;
  notes?: string;
};

export type EligibleReturnItem = {
  product_id: number;
  sku: string;
  name: string;
  sold_qty: number;
  already_returned: number;
  available_qty: number;
  unit_price: number;
};

export type CreateReturnPayload = {
  invoice_id: number;
  items: Array<{
    product_id: number;
    qty: number;
  }>;
  refund_method: string;
  notes?: string;
};

export type CreateReturnResult = {
  success: boolean;
  return_id: number;
  refund_amount: number;
};

export function getReturns(token: string) {
  return apiRequest<PaginatedItems<ReturnRow>>("/returns?limit=12", { token });
}

export function getEligibleReturnItems(token: string, invoiceId: number) {
  return apiRequest<EligibleReturnItem[]>(`/returns/invoice/${invoiceId}/eligible`, { token });
}

export function createReturn(token: string, payload: CreateReturnPayload) {
  return apiRequest<CreateReturnResult>("/returns", {
    method: "POST",
    token,
    body: payload
  });
}

File: app/models/returns.py
"""Returns schemas."""
from pydantic import BaseModel, Field
from typing import List, Optional


class ReturnItemRequest(BaseModel):
    product_id: int
    qty: int = Field(..., gt=0)


class CreateReturnRequest(BaseModel):
    invoice_id: int
    items: List[ReturnItemRequest]
    refund_method: str = "cash"
    notes: str = ""


class ReturnItemOut(BaseModel):
    id: int
    product_id: int
    product_name: str = ""
    qty: int
    unit_price: float
    line_total: float


class ReturnOut(BaseModel):
    id: int
    created_at: str
    original_invoice_id: int
    customer_id: Optional[int] = None
    user_id: int
    refund_method: str = "cash"
    refund_amount: float = 0.0
    branch_id: int = 1
    notes: str = ""
    items: List[ReturnItemOut] = []


class EligibleReturnItem(BaseModel):
    product_id: int
    sku: str
    name: str
    sold_qty: int
    already_returned: int
    available_qty: int
    unit_price: float
